network.go

```diff
@@ -135,3 +152,34 @@ func getIPFromHeader(r *http.Request, name string) net.IP {
 
 	return nil
 }
+
+func extractSubFromJWT(jwtString, secret string) (string, error) {
+	claims := &jwt.StandardClaims{}
+	token, err := jwt.ParseWithClaims(jwtString, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(secret), nil
+	})
+	if err != nil {
+		return "", err
+	}
+	if !token.Valid {
+		return "", http.ErrLineTooLong
+	}
+	fmt.Println(claims.Subject)
+	return fmt.Sprint([]byte(claims.Subject)), nil
+}
+
+func getAuthorizationToken(r *http.Request) string {
+	headerToken := r.Header.Get("Authorization")
+	if headerToken == "" {
+		return ""
+	}
+
+	// Verify the token format (Bearer <token>)
+	const bearer = "Bearer "
```

Comment:
Your code doesn't check if the content of `Authorization` starts with `Bearer`.
Also, you don't support if Bearer is in lowercase.